samcli/commands/pipeline/bootstrap/guided_context.py

```diff
@@ -33,81 +36,195 @@ def __init__(
 
-        if not self.artifacts_bucket_arn:
-            self.artifacts_bucket_arn = click.prompt(
-                "\nArtifacts bucket (S3 bucket to hold the AWS SAM build artifacts) [leave blank to create one]",
-                default="",
-                type=click.STRING,
-            )
-        if not self.image_repository_arn:
-            click.echo(
-                "\nIf your SAM template includes (or going to include) Lambda functions of Image package type, "
-                "then an ECR image repository is required. Should we create one?"
-            )
-            click.echo("\t1 - No, My SAM template won't include Lambda functions of Image package type")
-            click.echo("\t2 - Yes, I need help creating one")
-            click.echo("\t3 - I already have an ECR image repository")
-            choice = click.prompt(text="Choice", show_choices=False, type=click.Choice(["1", "2", "3"]))
-            if choice == "1":
-                self.create_image_repository = False
-            elif choice == "2":
-                self.create_image_repository = True
-            else:  # choice == "3"
-                self.create_image_repository = False
-                self.image_repository_arn = click.prompt("ECR image repository", type=click.STRING)
-
-        if not self.pipeline_ip_range:
-            click.echo("\nWe can deny requests not coming from a recognized IP address range.")
-            self.pipeline_ip_range = click.prompt(
-                "Pipeline IP address range (using CIDR notation) [leave blank if you don't know]",
-                default="",
-                type=click.STRING,
+        if self.artifacts_bucket_arn:
+            click.echo(f"Artifacts bucket ARN: {self.cloudformation_execution_role_arn}")
+        else:
+            self._prompt_artifacts_bucket()
+
+        if self.image_repository_arn:
+            click.echo(f"ECR image repository ARN: {self.image_repository_arn}")
+        else:
+            self._prompt_image_repository()
+
+        click.secho("[4] Security definition - OPTIONAL", bold=True)
+        if self.pipeline_ip_range:
+            click.echo(f"Pipeline IP address range: {self.pipeline_ip_range}")
+        else:
+            self._prompt_ip_range()
+
+        # Ask customers to confirm the inputs
+        while True:
+            inputs = self._get_user_inputs()
+            for i, (text, _) in enumerate(inputs):
+                click.secho(f"{i + 1}. {text}", fg="cyan")
+            edit_input = click.prompt(
+                text="Press enter to confirm the values above, or select an item to edit the value",
+                default="0",
+                show_choices=False,
+                show_default=False,
+                type=click.Choice(["0"] + [str(i + 1) for i in range(len(inputs))]),
             )
+            if int(edit_input):
+                inputs[int(edit_input) - 1][1]()
+            else:
+                break
```

Comment:
Shouldn't the input by this point be _guaranteed_ valid? i.e. `click` should force a valid response.